packages/flame/lib/src/components/router_component.dart

```diff
@@ -104,18 +104,68 @@ class RouterComponent extends Component {
     _adjustRoutesVisibility();
   }
 
+  /// Pops the current route and places the route with [name] on top of the
+  /// navigation stack.
+  ///
+  /// If the route is already in the stack, it will simply be moved to the top.
+  /// Otherwise the route will be mounted and added at the top. The route's
+  /// page will also start building if it hasn't been built before. If the
+  /// route is already on top of the stack, this method will do nothing.
+  ///
+  /// This method calls the [Route.didPush] callback for the newly activated
+  /// route and also calls the [Route.didPop] callback for the popped route.
+  void pushReplacementNamed(String name) {
+    final route = _resolveRoute(name);
+    if (route == currentRoute) {
+      return;
+    } else {
+      _popWithoutMinimumAssert();
+    }
+    if (_routeStack.contains(route)) {
+      _routeStack.remove(route);
+    } else {
+      add(route);
+    }
+    _routeStack.add(route);
+    _adjustRoutesOrder();
+    route.didPush(previousRoute);
+    _adjustRoutesVisibility();
+  }
+
   /// Puts a new [route] on top of the navigation stack.
   ///
   /// The route may also be given a [name], in which case it will be cached in
   /// the [routes] map under this name (if there was already a route with the
   /// same name, it will be overwritten).
   ///
   /// The method calls [Route.didPush] for this new route after it is added.
-  void pushRoute(Route route, {String name = ''}) {
-    route.name = name;
-    if (name.isNotEmpty) {
+  void pushRoute(Route route, {String? name}) {
+    if (name != null) {
+      route.name = name;
+      _routes[name] = route;
+    }
+    add(route);
+    _routeStack.add(route);
+    _adjustRoutesOrder();
+    route.didPush(previousRoute);
+    _adjustRoutesVisibility();
+  }
+
+  /// Pops the current route and puts a new [route] on top of the navigation
+  /// stack.
+  ///
+  /// The route may also be given a [name], in which case it will be cached in
+  /// the [routes] map under this name (if there was already a route with the
+  /// same name, it will be overwritten).
+  ///
+  /// The method calls [Route.didPush] for this new route after it is added and
+  /// also calls the [Route.didPop] callback for the popped route.
+  void pushReplacement(Route route, {String? name}) {
```

Comment:
(same here)

Reply:
And this one has this counter-part:
https://api.flutter.dev/flutter/widgets/Navigator/pushReplacement.html